carbon-design-system/carbon
# fix(button): update types hasIconOnly & kind !== danger

Closes #17036

[warnings visible when setting the wrong kind]
<img width="1728" alt="Screenshot 2024-08-26 at 10 55 22 AM" src="https://github.com/user-attachments/assets/cc5f7984-0b28-4d1d-8524-564fb90eca92">

#### Changelog

**Changed**

- update ButtonBaseProps `kind` to equal `hasIconOnly` ? `iconButtonKind` : `buttonKind`
- and updated Button.propTypes similarily to above^

#### Testing / Reviewing

- Go to storybook button confirm no warning errors
- go to storybook button -> icon button and change the kind to "danger" and see warnings
